images/base-python-adapter/swagger_server/server_logging.py

```diff
@@ -0,0 +1,86 @@
+#  Copyright 2023 VMware, Inc.
+#  SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+from configparser import ConfigParser
+from logging.handlers import RotatingFileHandler
+from typing import Optional
+
+log_handler: Optional[RotatingFileHandler] = None
+
+
+def setup_logging(
+    filename: str, file_count: int = 5, max_size: int = 10_485_760
+) -> None:
+    """
+    :param filename The name of the file to log to
+    :param file_count The total number of files to retain. Defaults to 5.
+    :param max_size The maximum size in bytes of each file before the file
+                automatically rotates to a new one. Defaults to '10_485_760' (10 MiB).
+    """
+    try:
+        global log_handler
+        log_handler = RotatingFileHandler(
+            os.path.join(os.sep, "var", "log", filename),
+            maxBytes=max_size,
+            backupCount=file_count,
+        )
+        logging.basicConfig(
+            format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
+            datefmt="%Y-%m-%d %H:%M:%S",
+            level=_get_default_log_level(),
+            handlers=[log_handler],
+        )
+        _set_log_levels()
+    except Exception:
+        logging.basicConfig(level=logging.CRITICAL + 1)
+
+
+def _get_default_log_level(default_level: int = logging.INFO) -> int:
+    default_level_name = logging.getLevelName(default_level)
+    log_config_file = os.path.join(os.sep, "var", "log", "loglevels.cfg")
+    config = ConfigParser()
+    if os.path.isfile(log_config_file):
```

Comment:
does this mean we just use defaults if someone deletes or renames the log file?

Reply:
and if so, what is the default behavior?

Reply:
This is specifically for the config file. If someone deletes the log file itself, it will just be re-created the next time logging happens with the same settings as before. If someone deletes the config file (or more commonly, if it's the first time the server/adapter is being run, since there isn't a config file to begin with), then the config file will be (re)created with the default settings (INFO)